common/types/mapper/thrift/replicator_test.go

```diff
@@ -0,0 +1,841 @@
+			desc:  "empty input test",
+			input: &types.FailoverMarkerAttributes{},
+		},
+		{
+			desc:  "nil input test",
+			input: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			thriftObj := FromFailoverMarkerAttributes(tc.input)
+			roundTripObj := ToFailoverMarkerAttributes(thriftObj)
+			assert.Equal(t, tc.input, roundTripObj)
+		})
+	}
+}
+
+func TestFailoverMarkers(t *testing.T) {
+	testCases := []struct {
+		desc  string
+		input *types.FailoverMarkers
+	}{
+		{
+			desc:  "non-nil input test",
+			input: &types.FailoverMarkers{FailoverMarkers: testdata.FailoverMarkerAttributesArray},
+		},
+		{
+			desc:  "empty input test",
+			input: &types.FailoverMarkers{},
+		},
+		{
+			desc:  "nil input test",
+			input: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			thriftObj := FromFailoverMarkers(tc.input)
+			roundTripObj := ToFailoverMarkers(thriftObj)
+			assert.Equal(t, tc.input, roundTripObj)
+		})
+	}
+}
+
+func TestAdminGetDLQReplicationMessagesRequest(t *testing.T) {
+	testCases := []struct {
+		desc  string
+		input *types.GetDLQReplicationMessagesRequest
+	}{
+		{
+			desc: "non-nil input test",
+			input: &types.GetDLQReplicationMessagesRequest{
+				TaskInfos: testdata.ReplicationTaskInfoArray,
+			},
+		},
+		{
+			desc:  "empty input test",
+			input: &types.GetDLQReplicationMessagesRequest{},
+		},
+		{
+			desc:  "nil input test",
+			input: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			thriftObj := FromAdminGetDLQReplicationMessagesRequest(tc.input)
+			roundTripObj := ToAdminGetDLQReplicationMessagesRequest(thriftObj)
+			assert.Equal(t, tc.input, roundTripObj)
+		})
+	}
+}
+
+func TestAdminGetDLQReplicationMessagesResponse(t *testing.T) {
+	testCases := []struct {
+		desc  string
+		input *types.GetDLQReplicationMessagesResponse
+	}{
+		{
+			desc: "non-nil input test",
+			input: &types.GetDLQReplicationMessagesResponse{
+				ReplicationTasks: testdata.ReplicationTaskArray,
+			},
+		},
+		{
+			desc:  "empty input test",
+			input: &types.GetDLQReplicationMessagesResponse{},
+		},
+		{
+			desc:  "nil input test",
+			input: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			thriftObj := FromAdminGetDLQReplicationMessagesResponse(tc.input)
+			roundTripObj := ToAdminGetDLQReplicationMessagesResponse(thriftObj)
+			assert.Equal(t, tc.input, roundTripObj)
+		})
+	}
+}
+
+func TestAdminGetDomainReplicationMessageRequest(t *testing.T) {
+	testCases := []struct {
+		desc  string
+		input *types.GetDomainReplicationMessagesRequest
+	}{
+		{
+			desc:  "non-nil input test",
+			input: &testdata.GetDomainReplicationMessagesRequest,
+		},
+		{
+			desc:  "empty input test",
+			input: &types.GetDomainReplicationMessagesRequest{},
+		},
+		{
+			desc:  "nil input test",
+			input: nil,
+		},
+	}
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
```

Comment:
a thing we've been kinda trying to shift towards / for minor test-boilerplate improvement, making the test container a `map[string]` can be convenient:
```suggestion
	testCases := map[string]*types.GetDomainReplicationMessagesRequest{
		"non-nil": &testdata.GetDomainReplicationMessagesRequest,
		"empty":   &types.GetDomainReplicationMessagesRequest{},
		"nil":     nil,
	}
	for name, tc := range testCases {
		t.Run(name, func(t *testing.T) {
```
we obviously mostly don't do this though, your `[]struct` is overwhelmingly the pattern in these tests, so this is not at all a request to change anything.  just an observation for anyone reading / reviewing / etc :)

Reply:
sure @Groxx will follow the map pattern for future tests.